### PR TITLE
fix(flow,ts): format `/* HTML */` templates

### LINKED ISSUE #5623: `html` in `.ts/.tsx` is not formatted
**Prettier 1.15.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEALGBbAG0wF5MB6AKkwAkAVAWQBlMKzMADAHgEIARAeQDCtAJoAFAKI58BAHwAdKJzpNMgxgEEAyluLyQUCAFoAVmkx4RRwQQ1ot+hVEyZlEjbycuXnehNoaatQaAEpa-nogAK4wAGZGAByOit6utACStIwSsvQAnpgwmQQSnGQZWTkp3r7+gQByGn6RACZwaGAATgCWAA4w3dD6mJCwCDCR+WoQ9TASULTJzq5k1O6e1ZwARhAteV41YrLUcAQEEJgA7hCdBC3cnABCIbKYtNjd5p80DIwArJhHhBugQ4J1egQAIbwAB0ZSO1R8WkEIXSYloB1Sl26UBaEEuMIA5pCSJhYlEoGABtBMAAKACUmGAmGJMIAjjDelE0NhaZDOoSonhxmhGQBfADcLMh7NJAG0ALpS1lEUgAaigcEumF40LgEsR3mJtIA5F04HqTQAaTAmgCqGiMAA0XS6jCJrbbITEICb6crIaa0AgWp6Tb1IYS4AA3bpav2GsodHr9TGuZGo9GYNCdMCRXAwXpoJBkMiXctEiAQQmgoyQqCQgh5AZgNAwyB4Mj1xvN7qtmFmYbaET1QQ6iQAMQkLyTXT6MAOZR2eycZRUjFk7BAVpAEH6gygaGQoH5nXxYn5CCPKEh0eBLW3IC2nUhYAA1nAYFoI2AcYTkDAnRRHAO64IQADqHzwGgP5wFoV7dAMsYwHkyDgGgR47jiwadDAYgvoSeCQsgsSNsGO5mAAHo8L7vp+WiQsKjA4nAJFkSBIBUVof6ggAilEEDwGxBDkSAEadDhaEob07Rzv0j69D0sDgd0LQwNgyAJAADDuikQMG4Evr0aGKe0YLRqxO6dHAbJRN01n4ZGRHCaJwZ4N0AFARxaA8XA-GCaxSCkSJHEwJCWwqWpGlIAATDugGQiCf6CBAeDOSghiao+3JwLQ4XXsFwZimKQA)
```sh
--parser typescript
```

**Input:**
```tsx
const html = /* HTML */ `<!DOCTYPE html>
<HTML CLASS="no-js mY-ClAsS">
  <HEAD>
    <META CHARSET="utf-8">
    <TITLE>My tITlE</TITLE>
    <META NAME="description" content="My CoNtEnT">
  </HEAD>
  <body>
    <P>Hello world!<BR> This is HTML5 Boilerplate.</P>
    <SCRIPT>
      window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
      ga('create', 'UA-XXXXX-Y', 'auto'); ga('send', 'pageview')
    </script>
    <SCRIPT src="https://www.google-analytics.com/analytics.js" ASYNC DEFER></script>
  </body>
</HTML>`
```

**Output:**
```tsx
const html = /* HTML */ `<!DOCTYPE html>
<HTML CLASS="no-js mY-ClAsS">
  <HEAD>
    <META CHARSET="utf-8">
    <TITLE>My tITlE</TITLE>
    <META NAME="description" content="My CoNtEnT">
  </HEAD>
  <body>
    <P>Hello world!<BR> This is HTML5 Boilerplate.</P>
    <SCRIPT>
      window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
      ga('create', 'UA-XXXXX-Y', 'auto'); ga('send', 'pageview')
    </script>
    <SCRIPT src="https://www.google-analytics.com/analytics.js" ASYNC DEFER></script>
  </body>
</HTML>`;

```

**Expected behavior:**
format html correctly like [JavaScript](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEALGBbAG0wF5MB6AKkwAkAVAWQBlMKzMADAHgEIARAeQDCtAJoAFAKI58BAHwAdKJzpNMgxgEEAyluLyQUCAFoAVmkx4RRwQQ1ot+hVEyZlEjbycuXnehNoaatQaAEpa-nogAK4wAGZGAByOit6utACStIwSsvQAnpgwmQQSnGQZWTkp3r7+gQByGn6RACZwaGAATgCWAA4w3dD6mJCwCDCR+WoQ9TASULTJzq5k1O6e1ZwARhAteV41YrLUcAQEEJgA7hCdBC3cnABCIbKYtNjd5p80DIwArJhHhBugQ4J1egQAIbwAB0ZSO1R8WkEIXSYloB1Sl26UBaEEuMIA5pCSJhYlEoGABtBMAAKACUmGAmGJMIAjjDelE0NhaZDOoSonhxmhGQBfADcLMh7NJAG0ALpS1lEUgAaigcEumF40LgEsR3mJtIA5F04HqTQAaTAmgCqGiMAA0XS6jCJrbbITEICb6crIaa0AgWp6Tb1IYS4AA3bpav2GsodHr9TGuZGo9GYNCdMCRXAwXpoJBkMiXctEiAQQmgoyQqCQgh5AZgNAwyB4Mj1xvN7qtmFmYbaET1QQ6iQAMQkLyTXT6MAOZR2eycZRUjFk7BAVpAEH6gygaGQoH5nXxYn5CCPKEh0eBLW3IC2nUhYAA1nAYFoI2AcYTkDAnRRHAO64IQADqHzwGgP5wFoV7dAMsYwHkyDgGgR47jiwadDAYgvoSeCQsgsSNsGO5mAAHo8L7vp+WiQsKjA4nAJFkSBIBUVof6ggAilEEDwGxBDkSAEadDhaFbJCWx5OcUCPr0PSwOB3QtDA2DIAkAAMO5KRAwbgS+vRoUp7RgtGrE7p0cBslE3Q2fhkZEcJonBng3QAUBHFoDxcD8YJrFIKRIkcTAMmqepmlIAATDugGQiCf6CBAeAuSghiao+3JwLQMnXiFwZimKQA)

### PR DESCRIPTION
**Pull request by ikatyang**
Fixes #5623

- [x] I’ve added tests to confirm my change works.
- (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/embed.js [hunk 1]
```diff
@@ -508,12 +510,9 @@ function hasLanguageComment(node, languageName) {
...
   // Also see ./clean.js
-  return (
-    node.leadingComments &&
-    node.leadingComments.some(
-      comment =>
-        comment.type === "CommentBlock" && comment.value === ` ${languageName} `
-    )
+  return hasLeadingComment(
+    node,
+    comment => isBlockComment(comment) && comment.value === ` ${languageName} `
```
Comment (j-f1) [inline src/language-js/embed.js]: `comment.value.trim() ===languageName`?

Comment (duailibe) [reply src/language-js/embed.js]: Unless you'd like to match the comment exactly with a single space

Comment (ikatyang) [reply src/language-js/embed.js]: https://github.com/prettier/prettier/pull/4395#commitcomment-28780334 😉

https://github.com/prettier/prettier/blob/3caa667240616cc582291a40d3fa3ae0131bb4f3/src/language-js/embed.js#L508-L511

Comment (j-f1) [reply src/language-js/embed.js]: Well look at me being all duplicative :)

**Pull request merged**